packages/connectors/src/metaMaskSDK.ts

```diff
@@ -0,0 +1,212 @@
+import { ConnectorNotFoundError } from './errors'
+import { InjectedConnector } from './injected'
+import { WindowProvider } from './types'
+
+import {
+  EventType,
+  MetaMaskSDK,
+  MetaMaskSDKOptions,
+  SDKProvider,
+} from '@metamask/sdk'
+import {
+  Address,
+  Chain,
+  ProviderRpcError,
+  ResourceUnavailableRpcError,
+  UserRejectedRequestError,
+  getAddress,
+} from 'viem'
+
+export type MetaMaskSDKConnectorOptions = {
+  // Keep both sdk and sdkOptions as some users might want to use their own pre-defined sdk instance
+  sdk?: MetaMaskSDK
```

Comment:
Lets remove the option to pass in `sdk` for now. Since Wagmi manages a few properties internally, think it makes sense to keep tighter control for the initial release. It's much easier to add the ability to pass in a `sdk` later than take it away. Once this is out in the wild and we receive feedback, happy to consider adding it back in.